httpx/dispatch/proxy_http.py

```diff
@@ -24,10 +23,9 @@
 logger = get_logger(__name__)
 
 
-class HTTPProxyMode(enum.Enum):
-    DEFAULT = "DEFAULT"
-    FORWARD_ONLY = "FORWARD_ONLY"
-    TUNNEL_ONLY = "TUNNEL_ONLY"
+DEFAULT_MODE = "DEFAULT"
+FORWARD_ONLY = "FORWARD_ONLY"
+TUNNEL_ONLY = "TUNNEL_ONLY"
```

Comment:
Dropping this enum is a breaking change — I believe `HTTPProxyMode` is documented under the proxy docs. Should we refrain from getting this change here, and see what we do with it later?

Reply:
Thanks! Switched to a retain-for-compat style, with warnings raised.

I think it's a reasonable change, right? Better to have a minimal API surface area, than expose configuration enums.

Reply:
Can roll back the change completely if you're unsure on it, mind.